[macOS] Upstream datalist suggestion dropdown changes and enable Liquid Glass background

lilyspiniolas opened the pull request:
#### 176ac0909ad4093fdee0d086d71f2d9aad959325
<pre>
[macOS] Upstream datalist suggestion dropdown changes and enable Liquid Glass background
<a href="https://bugs.webkit.org/show_bug.cgi?id=296206">https://bugs.webkit.org/show_bug.cgi?id=296206</a>
<a href="https://rdar.apple.com/156172489">rdar://156172489</a>

Reviewed by Aditya Keerthi.

Upstream changes to `WebDataListSuggestionsDropdownMac.mm` which enforce a corner
radius that matches native menus.

Additionally, use Liquid Glass for the background of the dropdown by using a
`NSGlassEffectView` rather than a `NSVisualEffectView` when Liquid Glass is
enabled.

* Source/WebKit/UIProcess/mac/WebDataListSuggestionsDropdownMac.mm:
(-[WKDataListSuggestionWindow initWithContentRect:styleMask:backing:defer:]):
(-[WKDataListSuggestionsController initWithInformation:inView:]):
(-[WKDataListSuggestionWindow adjustWindowIfNeeded]): Deleted.

Canonical link: <a href="https://commits.webkit.org/297629@main">https://commits.webkit.org/297629@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e13e4961605a7fc412ad775fb8d729ea52ad149

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112336 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32068 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22545 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/118415 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62704 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32720 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40631 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85336 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/36037 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115283 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26107 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101070 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65767 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25407 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62260 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95491 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19284 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121740 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39410 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29333 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94144 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39791 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97311 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93970 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24031 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39215 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17011 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/35460 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39298 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44786 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/38933 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42270 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40676 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/UIProcess/mac/WebDataListSuggestionsDropdownMac.mm:
```diff
@@ -144,12 +144,23 @@ - (id)initWithContentRect:(NSRect)contentRect styleMask:(NSUInteger)styleMask ba
+        _backdropView = adoptNS([[NSGlassEffectView alloc] initWithFrame:contentRect]);
+#endif
 
+    if (!liquidGlassEnabled) {
```
```suggestion
    if (! _backdropView) {
```

pxlcoder commented on Source/WebKit/UIProcess/mac/WebDataListSuggestionsDropdownMac.mm:
```diff
@@ -144,12 +144,23 @@ - (id)initWithContentRect:(NSRect)contentRect styleMask:(NSUInteger)styleMask ba
-    [_backdropView setMaterial:NSVisualEffectMaterialMenu];
-    [_backdropView setState:NSVisualEffectStateActive];
-    [_backdropView setBlendingMode:NSVisualEffectBlendingModeBehindWindow];
+    bool liquidGlassEnabled = false;
```
And then you don't need this variable.

lilyspiniolas committed:
[macOS] Upstream datalist suggestion dropdown changes and enable Liquid Glass background
https://bugs.webkit.org/show_bug.cgi?id=296206
rdar://156172489

Reviewed by Aditya Keerthi.

Upstream changes to `WebDataListSuggestionsDropdownMac.mm` which enforce a corner
radius that matches native menus.

Additionally, use Liquid Glass for the background of the dropdown by using a
`NSGlassEffectView` rather than a `NSVisualEffectView` when Liquid Glass is
enabled.

* Source/WebKit/UIProcess/mac/WebDataListSuggestionsDropdownMac.mm:
(-[WKDataListSuggestionWindow initWithContentRect:styleMask:backing:defer:]):
(-[WKDataListSuggestionsController initWithInformation:inView:]):
(-[WKDataListSuggestionWindow adjustWindowIfNeeded]): Deleted.

Canonical link: https://commits.webkit.org/297629@main
